weaviate/collections/grpc/query.py

```diff
@@ -301,22 +322,97 @@ def near_vector(
         group_by: Optional[_GroupBy] = None,
         generative: Optional[_Generative] = None,
         rerank: Optional[Rerank] = None,
-        target_vector: Optional[str] = None,
+        target_vector: Optional[TargetVectorJoinType] = None,
         return_metadata: Optional[_MetadataQuery] = None,
         return_properties: Optional[PROPERTIES] = None,
         return_references: Optional[REFERENCES] = None,
     ) -> search_get_pb2.SearchReply:
         if self._validate_arguments:
             _validate_input(
                 [
-                    _ValidateArgument([List], "near_vector", near_vector),
-                    _ValidateArgument([str, None], "target_vector", target_vector),
+                    _ValidateArgument([List, Dict], "near_vector", near_vector),
+                    _ValidateArgument(
+                        [str, None, List, _MultiTargetVectorJoin], "target_vector", target_vector
+                    ),
                 ]
             )
 
-        near_vector = _get_vector_v4(near_vector)
         certainty, distance = self.__parse_near_options(certainty, distance)
 
+        targets, target_vectors = self.__target_vector_to_grpc(target_vector)
+
+        if isinstance(near_vector, dict):
+            if targets is None or len(targets.target_vectors) != len(near_vector):
+                raise WeaviateInvalidInputError(
+                    "The number of target vectors must be equal to the number of vectors."
+                )
+
+            vector_per_target: Dict[str, bytes] = {}
+            for key, value in near_vector.items():
+                if (
+                    not isinstance(value, list)
+                    or len(value) == 0
+                    or not isinstance(value[0], get_args(NUMBER))
```

Comment:
I've not seen this pattern before. Does it allow the instance of `value[0]` to be checked against both `int` and `float` within the same `isinstance` call?

Reply:
I think chatGPT proposed it 😅 

> Does it allow the instance of value[0] to be checked against both int and float within the same isinstance call

Yes, AFAIK this just works in newer python versions, but not the old stuff we still have to support